src/test/java/org/fogbowcloud/manager/OrderStateTransitionerTest.java

```diff
@@ -84,73 +93,95 @@ public void testValidTransition() throws UnexpectedException {
     public void testDestinationListCannotBeFound() throws UnexpectedException {
+
+        // set up
         OrderState originState = OrderState.OPEN;
         OrderState destinationState = OrderState.SPAWNING;
 
         SharedOrderHolders ordersHolder = Mockito.mock(SharedOrderHolders.class);
-        when(ordersHolder.getOrdersList(originState))
-                .thenReturn(new SynchronizedDoublyLinkedList());
-
-        // destination list will fail to be found
-        when(ordersHolder.getOrdersList(destinationState)).thenReturn(null);
 
         PowerMockito.mockStatic(SharedOrderHolders.class);
-        given(SharedOrderHolders.getInstance()).willReturn(ordersHolder);
+        BDDMockito.given(SharedOrderHolders.getInstance()).willReturn(ordersHolder);
+
+        Mockito.when(ordersHolder.getOrdersList(originState))
+                .thenReturn(new SynchronizedDoublyLinkedList());
+
+        /** Destination list will fail to be found */
+        Mockito.when(ordersHolder.getOrdersList(destinationState)).thenReturn(null);
 
         Order order = createOrder(originState);
+
+        // exercise
         OrderStateTransitioner.transition(order, destinationState);
     }
 
+    // test case: When calling the transition() method and fail to remove an Order of the origin
+    // list, must not throws an UnexpectedException.
     @Test
     public void testOriginListRemovalFailure() {
+
+        // set up
         OrderState originState = OrderState.OPEN;
         OrderState destinationState = OrderState.SPAWNING;
 
-        SynchronizedDoublyLinkedList origin = 
-                Mockito.mock(SynchronizedDoublyLinkedList.class);
-
-        when(origin.removeItem(any(Order.class))).thenReturn(false);
-        
+        SynchronizedDoublyLinkedList origin = Mockito.mock(SynchronizedDoublyLinkedList.class);
         SynchronizedDoublyLinkedList destination = Mockito.mock(SynchronizedDoublyLinkedList.class);
 
         SharedOrderHolders ordersHolder = Mockito.mock(SharedOrderHolders.class);
-        when(ordersHolder.getOrdersList(originState)).thenReturn(origin);
-        when(ordersHolder.getOrdersList(destinationState)).thenReturn(destination);
 
         PowerMockito.mockStatic(SharedOrderHolders.class);
-        given(SharedOrderHolders.getInstance()).willReturn(ordersHolder);
+        BDDMockito.given(SharedOrderHolders.getInstance()).willReturn(ordersHolder);
+
+        Mockito.when(ordersHolder.getOrdersList(originState)).thenReturn(origin);
+        Mockito.when(ordersHolder.getOrdersList(destinationState)).thenReturn(destination);
+
+        Mockito.when(origin.removeItem(Mockito.any(Order.class))).thenReturn(false);
 
-        
         Order order = createOrder(originState);
-        
+
         try {
+
+            // exercise
             OrderStateTransitioner.transition(order, destinationState);
         } catch (UnexpectedException e) {
-            fail();
+
+            // verify
+            Assert.fail();
         }
```

Comment:
Shouldn't we verify the calls to the stub objects?

Reply:
Test was removed because it no longer has its initial function.